cmd/server-mux.go

```diff
@@ -258,18 +324,19 @@ func (m *ServerMux) ListenAndServeTLS(certFile, keyFile string) (err error) {
 func (m *ServerMux) ListenAndServe() error {
 	go m.handleServiceSignals()
 
-	listener, err := net.Listen("tcp", m.Server.Addr)
+	listeners, err := initListeners(m.Server.Addr, &tls.Config{})
 	if err != nil {
 		return err
 	}
 
-	listenerMux := &ListenerMux{Listener: listener, config: &tls.Config{}}
-
 	m.mu.Lock()
-	m.listener = listenerMux
+	m.listeners = listeners
 	m.mu.Unlock()
 
-	err = m.Server.Serve(listenerMux)
+	for _, listener := range listeners[:len(listeners)-1] {
+		go m.Server.Serve(listener)
+	}
+	err = m.Server.Serve(listeners[len(listeners)-1])
```

Comment:
If I remember the earlier discussion about this correctly, we should be checking for error when listening to any of the resolved addresses, and aborting even if any one of them errors out. [Case when a host name resolves to both a local interface and a non-local IP too]